controllers/dnspolicy_controller.go

```diff
@@ -199,5 +202,35 @@ func (r *DNSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				return gatewayEventMapper.MapToPolicy(ctx, object, &v1alpha1.DNSPolicy{})
 			}),
 		)
+	r.RegisterEventHandler(&DNSMetricsEventHandler{})
 	return ctrlr.Complete(r)
 }
+
+const (
+	dnsRecordNameLabel      = "dns_record_name"
+	dnsRecordNamespaceLabel = "dns_record_namespace"
+	dnsRecordEventTypeLabel = "dns_record_event_type"
+	dnsPolicyNameLabel      = "dns_policy_name"
+	dnsPolicyNamespaceLabel = "dns_policy_namespace"
```

Comment:
this is probably not needed. The dns record is always is the same ns as the policy

Reply:
this is for the dns policy ready metric.